frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java

```diff
@@ -79,8 +79,9 @@ public void ancestorAdded(AncestorEvent event) {
           val macheteFilePath = gitRepository.getMacheteFilePath();
           Runnable queueDiscoverOperation = () -> graphTable.queueDiscover(macheteFilePath, () -> {});
           val rediscoverSuggester = new RediscoverSuggester(gitRepository, queueDiscoverOperation);
-          graphTable.queueRepositoryUpdateAndModelRefresh(() -> rediscoverSuggester.perform());
+          graphTable.queueRepositoryUpdateAndModelRefresh(rediscoverSuggester::enqueueChecksAndPerformIfApplicable);
         }
+        removeAncestorListener(this);
```

Comment:
Hmmm is this change related to the PR? 🤔

Reply:
Anyway, I'd leave a comment in code on why it's needed as it's quite non-trivial

Reply:
Yes, it's related. Before this PR the rediscovery suggestion was not proposed because the last edit time of the .git/machete file was set. Now we have more operations before checking the date of the file and this is to ensure that we don't do them constantly.

Reply:
Okay, now we do the modification date checking first then check if all local branches are managed.